textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/nlpextrator/NlpExtractor.java

```diff
@@ -93,64 +132,108 @@ public ITuple getNextTuple() throws Exception {
+            ITuple returnTuple = Utils.getSpanTuple(sourceTuple.getFields(),
+                    spanList, returnSchema);
             sourceTuple = sourceOperator.getNextTuple();
             return returnTuple;
         }
     }
 
     /**
      * @param iField
-     * @return a List of spans of the extracted information
-     * @about This function takes an (TextField) IField and a String
-     * (the field's name) as input and uses the Stanford NLP package to process the field.
-     * It returns a list of spans
-     * In the returning span: Value -> the word itself
-     * Key   -> NE_Constant
-     * @overview Using the Stanford NLP package to process the textField value.
-     * First set up a pipeline of Annotators for the string: TokenizerAnnotator,
+     * @param fieldName
+     * @return
+     * @about This function takes an IField(TextField) and a String
+     * (the field's name) as input and uses the Stanford NLP package
+     * to process the field based on the input constant and flag.
+     * In the result spans, value represents the word itself
+     * and key represents the recognized constant
+     * @overview First set up a pipeline of Annotators based on the flag.
+     * If the flag is "NE_ALL", we set up the NamedEntityTagAnnotator,
+     * if it's "POS", then only PartOfSpeechAnnotator is needed.
+     * <p>
+     * The pipeline has to be this order: TokenizerAnnotator,
      * SentencesAnnotator, PartOfSpeechAnnotator, LemmaAnnotator and
-     * NamedEntityTagAnnotator. The order is mandatory because they have
-     * dependencies. After the pipeline, each token is wrapped as a CoreLabel
+     * NamedEntityTagAnnotator.
+     * <p>
+     * In the pipeline, each token is wrapped as a CoreLabel
      * and each sentence is wrapped as CoreMap. Each annotator adds its
      * annotation to the CoreMap(sentence) or CoreLabel(token) object.
      * <p>
-     * After the pipeline is done, scan each CoreLabel(token) for
-     * its NamedEntityAnnotation, if it's a valid value (not 'O'),
-     * then makes it a span and add to the return list.
-     * The Stanford NLP constants are mapped into the NE constants.
+     * After the pipeline, scan each CoreLabel(token) for its
+     * NamedEntityAnnotation or PartOfSpeechAnnotator depends on the flag
+     * <p>
+     * For each Stanford NLP annotation, get it's corresponding inputNlpConstant
+     * that used in this package, then check if it equals to the input constant.
+     * If yes, makes it a span and add to the return list.
+     * <p>
      * The NLP package has annotations for the start and end position of a token
      * and it perfectly matches the span design so we just use them.
-     *
+     * <p>
+     * For Example: With TextField value: "Microsoft, Google and Facebook are
+     * organizations while Donald Trump and Barack Obama are persons", with
+     * fieldName: Sentence1 and inputConstant is Organization. The flag would
```

Comment:
This sentence "The flag would set to "NE" by constructor." is not clear.  Rephrase it.